### PR TITLE
feat(atomic, commerce): create atomic-product-multi-value-text component

### PR DESCRIPTION
**Pull request by louis-bompart**
https://coveord.atlassian.net/browse/KIT-3440

Implement the `atomic-product-multi-value-text` component to render the values of a multi-value string field.

* **New Component**: Add `atomic-product-multi-value-text` component in `packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx`.
  - Combine functionalities from `atomic-product-text`, `atomic-result-text`, and `atomic-result-multi-value-text`.
  - Use `ProductContext` to get the product data.
  - Use `getFieldValueCaption` to get the field value caption.
  - Render the multi-value text field values.
* **Storybook**: Add a new story for `atomic-product-multi-value-text` in `packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.new.stories.tsx`.
  - Test rendering of multi-value text field values.
  - Test rendering with different `max-values-to-display` values.
  - Test rendering with custom slot values.

---

For more details, open the [Copilot Workspace session](https://copilot-workspace.githubnext.com/coveo/ui-kit?shareId=02336614-673d-4772-9196-21872fd10d36).

---
KIT-3440

### CODE REVIEW

**Review by fpbrault — APPROVED** [review]

File: packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+  /**
+   * The field that the component should use.
+   * The component will try to find this field in the `Product.raw` object unless it finds it in the `Product` object first.
+   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-commerce-interface` component.
+   */
+  @Prop({reflect: true}) public field!: string;
+
+  /**
+   * The maximum number of field values to display.
+   * If there are _n_ more values than the specified maximum, the last displayed value will be "_n_ more...".
```
[the remark this answers is not in the document]
Comment (fbeaudoincoveo) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:51]: Actually I think the description accurately reflects the behavior we should expect, but the component's behavior is flawed.

If I set 'maxValuesToDisplay' to 2 and there are 3 values available in total, I would definitely expect 2 values to be displayed, and a string saying "1 more..." to be rendered.

Similarly, if `maxValuesToDisplay` is set to 1 and there are 3 values available in total, we should display 1 value and a string saying "2 more..."

Right now, the latter works, but not the former.

This is because of what we do in the `getNumberOfValuesToDisplay` function:

```typescript
  private getNumberOfValuesToDisplay(values: string[]) {
    if (values.length <= this.maxValuesToDisplay) {
      return values.length;
    }
    if (this.maxValuesToDisplay < 2) {
      return this.maxValuesToDisplay;
    }
    return Math.min(values.length - 2, this.maxValuesToDisplay);
  }
```

if we suppose `values.length === 3` and `this.maxValuesToDisplay == 2`, then the function will return `Math.min(values.length - 2, this.maxValuesToDisplay)`, i.e., 1. This is not what we want.

File: packages/atomic/storybookUtils/product-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import {Decorator} from '@storybook/web-components';
+import {html, render} from 'lit';
+import {unsafeHTML} from 'lit/directives/unsafe-html.js';
+import type * as _ from '../src/components.js';
+
+export const wrapInProduct = (): {
```
Comment (alexprudhomme) [inline packages/atomic/storybookUtils/product-wrapper.tsx:6]: This file is not used !

**Review by alexprudhomme — APPROVED** [review]

File: packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+
+  private get facetSelectedValues() {
+    return this.breadcrumbManager.state.facetBreadcrumbs
+      .filter((facet) => facet.field === this.field)
+      .reduce(
+        (values, facet) => [
+          ...values,
+          ...facet.values.map(({value}) => (value as RegularFacetValue).value),
+        ],
+        [] as string[]
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:106]: I'm wondering if we should put [`no-array-reduce` ](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-reduce.md)in our eslint rules 😄 
[Perf.link](https://perf.link/#eyJpZCI6Im5vcm5ocjgxZjltIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImxldCBhcnIgPSBuZXcgQXJyYXkoMTAwKTtcbmNvbnN0IGx1Y2t5S2V5ID0gJ2ZvbydcbmxldCBsdWNreUluZGljZXMgPSBuZXcgVWludDMyQXJyYXkoNCk7XG5zZWxmLmNyeXB0by5nZXRSYW5kb21WYWx1ZXMobHVja3lJbmRpY2VzKTtcbmx1Y2t5SW5kaWNlcyA9IEFycmF5LmZyb20obHVja3lJbmRpY2VzKS5tYXAoaW5kZXggPT4gaW5kZXggJSBhcnIubGVuZ3RoKTtcblxuY29uc3QgY29udHJvbFZhbHVlcyA9IFtdO1xuXG5mb3IodmFyIGlkeCA9IDAgOyBpZHg8YXJyLmxlbmd0aDsgaWR4KysgKSB7XG4gIGFycltpZHhdPSB7XG4gICAgIGZpZWxkOiBsdWNreUluZGljZXMuaW5jbHVkZXMoaWR4KSA%2FIGx1Y2t5S2V5IDogJ2JhcicsXG4gICAgIHZhbHVlczogQXJyYXkuZnJvbShzZWxmLmNyeXB0by5nZXRSYW5kb21WYWx1ZXMobmV3IFVpbnQzMkFycmF5KDEwMCkpKVxuICAgICAubWFwKCh2YWx1ZSk9Pih7dmFsdWV9KSlcbiAgIH1cbiAgaWYoYXJyW2lkeF0uZmllbGQgPT09IGx1Y2t5S2V5KSB7XG4gICAgY29udHJvbFZhbHVlcy5wdXNoKC4uLihhcnJbaWR4XS52YWx1ZXMubWFwKCh2YWx1ZSk9PnZhbHVlLnZhbHVlKSkpO1xuICB9XG59IiwidGVzdHMiOlt7Im5hbWUiOiJPcHRpbWl6ZWQtZm9yLWkiLCJjb2RlIjoidmFyIHNlbGVjdGVkVmFsdWVzID0gW107XG52YXIgbGVuID0gYXJyLmxlbmd0aDtcbmZvcih2YXIgaWR4ID0gMCA7IGlkeDxsZW47IGlkeCsrICkge1xuICBpZihhcnJbaWR4XS5maWVsZCA9PT0gbHVja3lLZXkpIHtcbiAgICBzZWxlY3RlZFZhbHVlcy5wdXNoKC4uLmFycltpZHhdLnZhbHVlcy5tYXAoKHZhbHVlKSA9PiB2YWx1ZS52YWx1ZSkpO1xuICB9XG59XG4vL2NvbnNvbGUuYXNzZXJ0KEpTT04uc3RyaW5naWZ5KHNlbGVjdGVkVmFsdWVzKT09PUpTT04uc3RyaW5naWZ5KGNvbnRyb2xWYWx1ZXMpKSIsInJ1bnMiOlsxMzAwMCw1MDAwLDEyMDAwLDI1MDAwLDI1MDAwLDIwMDAsNDAwMDAsMjQwMDAsMjYwMDAsMjUwMDAsMjMwMDAsMjYwMDAsNjAwMCw0MDAwMCwyMDAwMCwyNTAwMCwzMDAwLDI1MDAwLDI1MDAwLDIxMDAwLDEwMDAwLDIwMDAsMTcwMDAsMjYwMDAsMTEwMDAsMTQwMDAsMTYwMDAsMTcwMDAsMjYwMDAsMTQwMDAsMjYwMDAsNzAwMCwxNjAwMCw3MDAwLDIwMDAsMTgwMDAsMTkwMDAsMjYwMDAsMjYwMDAsMjAwMDAsMjMwMDAsNDAwMCwyNTAwMCw1MDAwLDE3MDAwLDM5MDAwLDkwMDAsMzkwMDAsMTAwMCwzMjAwMCwyNTAwMCwxMTAwMCw0MDAwLDEwMDAsNDAwMCwyNTAwMCw3MDAwLDI1MDAwLDI0MDAwLDE4MDAwLDIwMDAwLDI2MDAwLDEzMDAwLDYwMDAsMjYwMDAsMTAwMCwyNTAwMCwyMzAwMCwyNTAwMCwyNTAwMCwyNTAwMCw3MDAwLDE1MDAwLDI1MDAwLDE4MDAwLDE4MDAwLDQwMDAsMjMwMDAsMjUwMDAsMjYwMDAsOTAwMCwyNjAwMCwxMzAwMCwyNTAwMCw5MDAwLDI1MDAwLDEzMDAwLDkwMDAsMjUwMDAsMjUwMDAsMTcwMDAsOTAwMCw1MDAwLDIyMDAwLDI2MDAwLDMxMDAwLDI1MDAwLDIxMDAwLDI0MDAwLDI2MDAwXSwib3BzIjoxODMwMH0seyJuYW1lIjoiT3Zlci1PcHRpbWl6ZWQtZm9yLWkiLCJjb2RlIjoidmFyIHNlbGVjdGVkVmFsdWVzID0gW107XG52YXIgbGVuID0gYXJyLmxlbmd0aDtcbmZvcih2YXIgaWR4ID0gMCA7IGlkeDxsZW47IGlkeCsrICkge1xuICB2YXIgZmFjZXQgPSBhcnJbaWR4XVxuICBpZihmYWNldC5maWVsZCA9PT0gbHVja3lLZXkpIHtcbiAgICBzZWxlY3RlZFZhbHVlcy5wdXNoKC4uLmZhY2V0LnZhbHVlcy5tYXAoKHZhbHVlKSA9PiB2YWx1ZS52YWx1ZSkpO1xuICB9XG59XG4vL2NvbnNvbGUuYXNzZXJ0KEpTT04uc3RyaW5naWZ5KHNlbGVjdGVkVmFsdWVzKT09PUpTT04uc3RyaW5naWZ5KGNvbnRyb2xWYWx1ZXMpKSIsInJ1bnMiOlsyMDAwMCwyNTAwMCwxMTAwMCwyNTAwMCwyNTAwMCwyNTAwMCwyNTAwMCwxNzAwMCwyMjAwMCwxMjAwMCw1MDAwLDI2MDAwLDI1MDAwLDQwMDAsMjUwMDAsMTQwMDAsMTYwMDAsMjIwMDAsMTgwMDAsMjYwMDAsMjYwMDAsMjUwMDAsMTcwMDAsMjUwMDAsMTAwMDAsMjYwMDAsMjAwMCwxMzAwMCwyNTAwMCwyNTAwMCwyNjAwMCwyNjAwMCwyNDAwMCw3MDAwLDI1MDAwLDEyMDAwLDI1MDAwLDI1MDAwLDIwMDAwLDEwMDAsMjEwMDAsMjYwMDAsMjUwMDAsMjYwMDAsMTgwMDAsMTgwMDAsOTAwMCw1MDAwLDI1MDAwLDIyMDAwLDI2MDAwLDkwMDAsMjYwMDAsMjUwMDAsMjYwMDAsMTIwMDAsMTAwMCwyMzAwMCwyMDAwMCwxMTAwMCwxNTAwMCwyMjAwMCw5MDAwLDI1MDAwLDQwMDAwLDI1MDAwLDQwMDAwLDE1MDAwLDIwMDAwLDI1MDAwLDI1MDAwLDEwMDAsMjUwMDAsMjQwMDAsMjYwMDAsNzAwMCw1MDAwLDI1MDAwLDI1MDAwLDE2MDAwLDI1MDAwLDMwMDAsNTAwMCwyNjAwMCwzMDAwLDMwMDAsMjUwMDAsMjUwMDAsMjYwMDAsMTkwMDAsMjUwMDAsMTAwMCwyNTAwMCw0MDAwLDI1MDAwLDMwMDAsMjUwMDAsMTUwMDAsMjEwMDAsMjUwMDBdLCJvcHMiOjE4OTEwfSx7Im5hbWUiOiJmb3ItaSIsImNvZGUiOiJ2YXIgc2VsZWN0ZWRWYWx1ZXMgPSBbXTtcbmZvcih2YXIgaWR4ID0gMCA7IGlkeDxhcnIubGVuZ3RoOyBpZHgrKyApIHtcbiAgaWYoYXJyW2lkeF0uZmllbGQgPT09IGx1Y2t5S2V5KSB7XG4gICAgc2VsZWN0ZWRWYWx1ZXMucHVzaCguLi5hcnJbaWR4XS52YWx1ZXMubWFwKCh2YWx1ZSkgPT4gdmFsdWUudmFsdWUpKTtcbiAgfVxufVxuLy9jb25zb2xlLmFzc2VydChKU09OLnN0cmluZ2lmeShzZWxlY3RlZFZhbHVlcyk9PT1KU09OLnN0cmluZ2lmeShjb250cm9sVmFsdWVzKSkiLCJydW5zIjpbODAwMCwyMTAwMCw4MDAwLDEzMDAwLDI2MDAwLDI2MDAwLDIwMDAsNDAwMDAsMTgwMDAsMjYwMDAsODAwMCwyMzAwMCwzNzAwMCwzMDAwLDkwMDAsMzAwMCwyMDAwLDI1MDAwLDMwMDAsMjAwMDAsNzAwMCwxMzAwMCwyNTAwMCwxMDAwLDI1MDAwLDI1MDAwLDI0MDAwLDIyMDAwLDQwMDAsMjEwMDAsMTYwMDAsMjYwMDAsMjYwMDAsMjYwMDAsMTcwMDAsMjYwMDAsMjUwMDAsMTcwMDAsNjAwMCwxNzAwMCwyMDAwLDE2MDAwLDIzMDAwLDkwMDAsMTcwMDAsMzAwMCw4MDAwLDI1MDAwLDEzMDAwLDIxMDAwLDE1MDAwLDI2MDAwLDIyMDAwLDE4MDAwLDI2MDAwLDIwMDAsMjUwMDAsMjQwMDAsNjAwMCw0MDAwMCwyNTAwMCw3MDAwLDI1MDAwLDI2MDAwLDIxMDAwLDIzMDAwLDI1MDAwLDIwMDAsMTYwMDAsMTcwMDAsMTMwMDAsMjMwMDAsMzAwMCwxNTAwMCwxOTAwMCwxNTAwMCw0MDAwMCwyMDAwMCwyMjAwMCw4MDAwLDE2MDAwLDIzMDAwLDEwMDAsMTQwMDAsMTIwMDAsMjEwMDAsNDEwMDAsMjAwMDAsMzIwMDAsMTMwMDAsNTAwMCwyMzAwMCwyMTAwMCwyNjAwMCwyMTAwMCwyNTAwMCwxOTAwMCwxNTAwMCwxOTAwMCwyMDAwMF0sIm9wcyI6MTc4MzB9LHsibmFtZSI6ImZvci1vZiIsImNvZGUiOiJ2YXIgc2VsZWN0ZWRWYWx1ZXMgPSBbXTtcbmZvcihjb25zdCBmYWNldCBvZiBhcnIpIHtcbiAgaWYoZmFjZXQuZmllbGQgPT09IGx1Y2t5S2V5KSB7XG4gICAgc2VsZWN0ZWRWYWx1ZXMucHVzaCguLi5mYWNldC52YWx1ZXMubWFwKCh2YWx1ZSkgPT4gdmFsdWUudmFsdWUpKTtcbiAgfVxufVxuLy9jb25zb2xlLmFzc2VydChKU09OLnN0cmluZ2lmeShzZWxlY3RlZFZhbHVlcyk9PT1KU09OLnN0cmluZ2lmeShjb250cm9sVmFsdWVzKSkiLCJydW5zIjpbMTcwMDAsMTcwMDAsNzAwMCw1MDAwLDE5MDAwLDE5MDAwLDIyMDAwLDIzMDAwLDI0MDAwLDEwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMTUwMDAsMjAwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMTgwMDAsMjQwMDAsMjIwMDAsMjIwMDAsMjkwMDAsMjMwMDAsMTkwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMjIwMDAsMjMwMDAsMjMwMDAsMTQwMDAsMjQwMDAsMjAwMDAsNjAwMCw1MDAwLDE4MDAwLDEwMDAsMzAwMCwzMDAwLDEyMDAwLDI0MDAwLDEwMDAsMjYwMDAsNTAwMCwxMDAwLDIwMDAsMTAwMCwyMzAwMCw4MDAwLDIzMDAwLDIzMDAwLDUwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMjQwMDAsODAwMCwyMzAwMCwyMzAwMCwyMzAwMCwyMzAwMCw5MDAwLDIzMDAwLDIwMDAwLDE1MDAwLDIwMDAwLDIzMDAwLDI0MDAwLDkwMDAsOTAwMCwyMzAwMCw4MDAwLDE0MDAwLDM1MDAwLDMwMDAsMjAwMCwyMzAwMCwxNzAwMCwyODAwMCwyMzAwMCwyMzAwMCwxNjAwMCw0MDAwLDE5MDAwLDIwMDAwLDYwMDAsNTAwMCwyMzAwMCwxMTAwMCw5MDAwLDIwMDAwLDMwMDAsMTkwMDAsMTAwMCwyMzAwMCw1MDAwLDEwMDBdLCJvcHMiOjE2NDQwfSx7Im5hbWUiOiJmbGF0LW1hcCIsImNvZGUiOiJ2YXIgc2VsZWN0ZWRWYWx1ZXMgPSBhcnIuZmxhdE1hcChmYWNldCA9PiB7XG4gIGlmKGZhY2V0LmZpZWxkICE9PSBsdWNreUtleSkge1xuICAgIHJldHVybiBbXTtcbiAgfVxuICByZXR1cm4gZmFjZXQudmFsdWVzLm1hcCgodmFsdWUpID0%2BIHZhbHVlLnZhbHVlKTtcbn0pXG4vL2NvbnNvbGUuYXNzZXJ0KEpTT04uc3RyaW5naWZ5KHNlbGVjdGVkVmFsdWVzKT09PUpTT04uc3RyaW5naWZ5KGNvbnRyb2xWYWx1ZXMpKSIsInJ1bnMiOlsxNDAwMCwyMDAwMCw4MDAwLDE2MDAwLDIyMDAwLDIzMDAwLDYwMDAsMjAwMCwxMDAwMCwyMzAwMCwxMDAwLDIwMDAsMjAwMCwxNjAwMCwxNDAwMCwyMDAwLDE0MDAwLDEwMDAsNDAwMCw0MDAwLDIzMDAwLDEwMDAwLDQwMDAsMjYwMDAsOTAwMCwyMjAwMCwxNDAwMCwxMDAwMCwyMDAwLDIwMDAwLDYwMDAsMTYwMDAsMjMwMDAsMjMwMDAsOTAwMCwyMDAwMCwyMzAwMCwxMjAwMCwxMzAwMCw0MDAwLDIzMDAwLDkwMDAsMjMwMDAsMjMwMDAsMjMwMDAsMTgwMDAsMjMwMDAsMjIwMDAsMTQwMDAsMTYwMDAsMjAwMDAsMTIwMDAsMjAwMDAsMjAwMCwyMzAwMCwxMDAwLDMyMDAwLDIwMDAwLDIwMDAsMTUwMDAsODAwMCwyMzAwMCwyMzAwMCwyMTAwMCwyMzAwMCwyMzAwMCw4MDAwLDE2MDAwLDMwMDAsMjIwMDAsMTUwMDAsMjMwMDAsMTIwMDAsMjMwMDAsMTQwMDAsNjAwMCwyMTAwMCwxODAwMCwyMjAwMCwxNjAwMCwyMzAwMCwyMDAwLDEyMDAwLDMyMDAwLDIzMDAwLDE4MDAwLDIzMDAwLDE4MDAwLDIxMDAwLDUwMDAsMjMwMDAsMjAwMDAsMzMwMDAsOTAwMCw3MDAwLDIzMDAwLDIxMDAwLDIwMDAsNjAwMCwxODAwMF0sIm9wcyI6MTUxMzB9LHsibmFtZSI6ImZpbHRlci1yZWR1Y2UiLCJjb2RlIjoidmFyIHNlbGVjdGVkVmFsdWVzID0gYXJyXG4gICAgICAuZmlsdGVyKChmYWNldCkgPT4gZmFjZXQuZmllbGQgPT09IGx1Y2t5S2V5KVxuICAgICAgLnJlZHVjZShcbiAgICAgICAgKHZhbHVlcywgZmFjZXQpID0%2BIFtcbiAgICAgICAgICAuLi52YWx1ZXMsXG4gICAgICAgICAgLi4uZmFjZXQudmFsdWVzLm1hcCgodmFsdWUpID0%2BIHZhbHVlLnZhbHVlKSxcbiAgICAgICAgXSxcbiAgICAgICAgW11cbiAgICAgICk7XG4vL2NvbnNvbGUuYXNzZXJ0KEpTT04uc3RyaW5naWZ5KHNlbGVjdGVkVmFsdWVzKT09PUpTT04uc3RyaW5naWZ5KGNvbnRyb2xWYWx1ZXMpKSIsInJ1bnMiOls4MDAwLDE2MDAwLDE0MDAwLDYwMDAsODAwMCw2MDAwLDQwMDAsMTYwMDAsMzAwMCwxNjAwMCw5MDAwLDIzMDAwLDEwMDAsMzAwMCwxNjAwMCw5MDAwLDE2MDAwLDE2MDAwLDYwMDAsMTYwMDAsNzAwMCw0MDAwLDE2MDAwLDEwMDAsMTYwMDAsMjAwMCw3MDAwLDE2MDAwLDEyMDAwLDIzMDAwLDEyMDAwLDE1MDAwLDYwMDAsOTAwMCwyMDAwLDE2MDAwLDMwMDAsMTAwMCwxNjAwMCwxNjAwMCwxMDAwMCwxNjAwMCwxMzAwMCwzMDAwLDE2MDAwLDE0MDAwLDEwMDAsMjAwMCwxMTAwMCwxNjAwMCwxMDAwLDcwMDAsMTYwMDAsMjAwMCwxNzAwMCw2MDAwLDcwMDAsMjAwMCwxNjAwMCwxNTAwMCwyMDAwLDE0MDAwLDExMDAwLDYwMDAsMjEwMDAsMTAwMCwxNjAwMCwxNjAwMCw1MDAwLDcwMDAsMTUwMDAsMTYwMDAsMTUwMDAsMjAwMCwxNjAwMCw2MDAwLDExMDAwLDMwMDAsMTAwMCwzMDAwLDgwMDAsMzAwMCwxMjAwMCwxNjAwMCw3MDAwLDE2MDAwLDE0MDAwLDEwMDAsMTAwMCw0MDAwLDE2MDAwLDExMDAwLDEyMDAwLDEyMDAwLDE2MDAwLDcwMDAsMTYwMDAsMTYwMDAsMzAwMCwxNjAwMF0sIm9wcyI6OTk3MH1dLCJ1cGRhdGVkIjoiMjAyNC0xMC0xNlQwMjoyNTo1Ny43MzVaIn0%3D)

TL;DR: Filter-reduce is 50% less efficient. (Well, we can also discuss readability, but that's subjective.)

To be clear, we should not unduly sacrifice readability for per and vice versa. But we ought to strike a good balance.
IMHO, filter-reduce is a irk solution: not so readable, definitely less performant.

A good listen to: https://www.youtube.com/watch?v=qaGjS7-qWzg

[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:106]: Also:
https://github.com/coveo/ui-kit/commit/7cab153c467336dc4a9fd9e67f1fa974c55cd3fd
image
XKCD
https://xkcd.coom/1421

[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:106]: Note to self: Should this kind of shenanigans be a selector

File: packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+  private updateSortedValues() {
+    const allValues = this.productValues;
+    if (allValues === null) {
+      this.sortedValues = null;
+      return;
+    }
+    const allValuesSet = new Set(allValues);
+    const firstValues = this.facetSelectedValues.filter((value) =>
+      allValuesSet.has(value)
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:118]: Yes: [perf](https://perf.link/#eyJpZCI6Inh6MHkxamVtY2ZmIiwidGl0bGUiOiJGaW5kIHRoZSBmaXJzdCBpbnN0YW5jZSBvZiB2YWx1ZXMgaW4gYW4gYXJyYXkiLCJiZWZvcmUiOiJsZXQgYWxsVmFsdWVzID0gQXJyYXkuZnJvbShzZWxmLmNyeXB0by5nZXRSYW5kb21WYWx1ZXMobmV3IFVpbnQ4QXJyYXkoMTBlMykpKTtcbmxldCBsdWNreUluZGljZXMgPSBBcnJheS5mcm9tKHNlbGYuY3J5cHRvLmdldFJhbmRvbVZhbHVlcyhuZXcgVWludDhBcnJheSgxMCkpKS5tYXAodmFsPT52YWwlYWxsVmFsdWVzLmxlbmd0aCk7XG5sZXQgc2VsZWN0ZWRWYWx1ZXMgPSBsdWNreUluZGljZXMubWFwKGlkeD0%2BYWxsVmFsdWVzW2lkeF0pIiwidGVzdHMiOlt7Im5hbWUiOiJUZXN0IENhc2UiLCJjb2RlIjoiICAgY29uc3QgZmlyc3RWYWx1ZXMgPSBzZWxlY3RlZFZhbHVlcy5maWx0ZXIoKHZhbHVlKSA9PiBhbGxWYWx1ZXMuaW5jbHVkZXModmFsdWUpKTsiLCJydW5zIjpbMTIzMzAwMCw0MjUwMDAsNjc0MDAwLDE1MDUwMDAsOTM2MDAwLDc1ODAwMCw4NzMwMDAsMzY0MDAwLDgwNTAwMCw0NzcwMDAsOTQzMDAwLDEyMjEwMDAsNjM2MDAwLDI4NzAwMCwxMTA0MDAwLDI1MTkwMDAsMjUyMTAwMCwyMTIwMDAsODA2MDAwLDUwMDAsMjE5MTAwMCw2NzIwMDAsODIxMDAwLDExNTYwMDAsMTAyNzAwMCwxMTM0MDAwLDE2MTUwMDAsMTcwNDAwMCwzMDAwLDYwNjAwMCw4OTYwMDAsMjQwMDAwMCw1MjAwMCwxMzQ3MDAwLDg5NjAwMCw1OTEwMDAsNzEzMDAwLDIzMzkwMDAsNTIwMDAsMjcwMDAsMjEyMzAwMCwzMDcwMDAsNTQyMDAwLDIyMTMwMDAsMTk5MjAwMCwxMjc1MDAwLDE1MjgwMDAsODY1MDAwLDE5NTcwMDAsNjEyMDAwLDYzMjAwMCwxMjU3MDAwLDE4NzIwMDAsMjc2MDAwLDIwMDAsMTU5MzAwMCwyMDQwMDAsMTIwNzAwMCw2NzkwMDAsNTkwMDAsMTk5MDAwMCwyNzM4MDAwLDQxMjAwMCwxNTQwMDAsMTI1NzAwMCwxNjQ2MDAwLDE0NzAwMDAsMTgyMjAwMCwxMDA0MDAwLDIyMDAwMCwyNzgwMDAsNzAwMDAwLDE0MzIwMDAsMTc5NzAwMCwxMDAwLDExMTgwMDAsMTIyMTAwMCwxOTQ5MDAwLDI2ODUwMDAsNTIwMDAsODkyMDAwLDI4NDAwMCwyMzcyMDAwLDEwMTQwMDAsNzUyMDAwLDY1MjAwMCwyMDY1MDAwLDc3OTAwMCwxNDQxMDAwLDc4ODAwMCwxMjE3MDAwLDEyMjEwMDAsMTEzMDAwLDMwNDAwMCwyNTgwMDAsNTQ5MDAwLDQ5MjAwMCwyNjMwMDAwLDI2MTMwMDAsMTEzMDAwXSwib3BzIjoxMDQyMzYwfSx7Im5hbWUiOiJTZXQgdHJpY2siLCJjb2RlIjoiY29uc3QgYWxsVmFsdWVzU2V0ID0gbmV3IFNldChhbGxWYWx1ZXMpO1xuY29uc3QgZmlyc3RWYWx1ZXMgPSBzZWxlY3RlZFZhbHVlcy5maWx0ZXIoKHZhbHVlKSA9PiBhbGxWYWx1ZXNTZXQuaGFzKHZhbHVlKSk7XG4gICIsInJ1bnMiOlszMDAwLDEwMDAsNjAwMCw0MDAwLDIwMDAsMTAwMCwyMDAwLDgwMDAsMjAwMCwxMDAwLDMwMDAsMzAwMCwxMDAwLDEwMDAsMjAwMCw3MDAwLDEwMDAsMTAwMCwyMDAwLDEwMDAsNjAwMCwxMDAwLDIwMDAsMzAwMCwyMDAwLDIwMDAsNTAwMCw0MDAwLDQwMDAsMzAwMCwyMDAwLDYwMDAsMTAwMCw1MDAwLDIwMDAsMTAwMCwzMDAwLDUwMDAsNzAwMCw0MDAwLDIwMDAsMjAwMCwxMDAwLDUwMDAsNTAwMCwzMDAwLDMwMDAsNzAwMCw2MDAwLDIwMDAsMTAwMCw1MDAwLDUwMDAsNjAwMCw2MDAwLDMwMDAsNzAwMCwyMDAwLDIwMDAsNzAwMCw1MDAwLDYwMDAsNzAwMCwyMDAwLDMwMDAsNTAwMCw0MDAwLDYwMDAsMjAwMCw2MDAwLDIwMDAsNDAwMCwzMDAwLDQwMDAsMjAwMCwzMDAwLDMwMDAsNjAwMCwyMDAwLDEwMDAsMjAwMCw3MDAwLDQwMDAsMzAwMCwyMDAwLDUwMDAsNTAwMCwyMDAwLDYwMDAsMTAwMCwzMDAwLDIwMDAsMTAwMCwyMDAwLDIwMDAsMTAwMCw4MDAwLDYwMDAsNzAwMCwxMDAwXSwib3BzIjozNDYwfV0sInVwZGF0ZWQiOiIyMDI0LTEwLTE2VDAyOjUzOjMwLjQwNloifQ%3D%3D)

File: packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    );
+    this.sortedValues = Array.from(
+      allValues.reduce((set, value) => set.add(value), new Set(firstValues))
+    );
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:122]: smart 👍

File: packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+   * The component will try to find this field in the `Product.additionalFields` object unless it finds it in the `Product` object first.
+   * Make sure this field is present in the `fieldsToInclude` property of the `atomic-commerce-interface` component.
+   */
+  @Prop({reflect: true}) public field!: string;
+
+  /**
+   * The maximum number of field values to display.
+   * If there are _n_ more values than the specified maximum, the last displayed value will be "_n_ more...".
+   */
+  @Prop({reflect: true}) public maxValuesToDisplay = 3;
```
[the remark this answers is not in the document]
Comment (louis-bompart) [reply packages/atomic/src/components/commerce/product-template-components/atomic-product-multi-value-text/atomic-product-multi-value-text.tsx:53]: https://github.com/coveo/ui-kit/blob/4fb7fd5eddf753fd27932dbcc6442ac6e1f4c3d4/packages/atomic/src/components/search/result-template-components/atomic-result-multi-value-text/atomic-result-multi-value-text.tsx#L116-L117
checked here 👍
